portal/v2/src/ClusterDetailListComponents/Statistics/GraphOptionsComponent.tsx

```diff
@@ -0,0 +1,220 @@
+        case "6h":
+          props.setDuration("12h")
+          break
+        case "12h":
+          props.setDuration("1d")
+          break
+        case "1d":
+          props.setDuration("2d")
+          break
+        case "2d":
+          props.setDuration("1w")
+          break
+        case "1w":
+          props.setDuration("2w")
+          break
+        case "2w":
+          props.setDuration("4w")
+          break
+        case "4w":
+          props.setDuration("8w")
+          break
+      }
+    }
+    const _decreaseDuration = () => {
+      switch(props.duration) {
+        case "5m":
+          props.setDuration("1m")
+          break
+        case "10m":
+          props.setDuration("5m")
+          break
+        case "30m":
+          props.setDuration("10m")
+          break
+        case "1h":
+          props.setDuration("30m")
+          break
+        case "2h":
+          props.setDuration("1h")
+          break
+        case "6h":
+          props.setDuration("2h")
+          break
+        case "12h":
+          props.setDuration("6h")
+          break
+        case "1d":
+          props.setDuration("12h")
+          break
+        case "2d":
+          props.setDuration("1d")
+          break
+        case "1w":
+          props.setDuration("2d")
+          break
+        case "2w":
+          props.setDuration("1w")
+          break
+        case "4w":
+          props.setDuration("2w")
+          break
+        case "8w":
+          props.setDuration("4w")
+          break
+      } 
+    }
+    const durationStyle: Partial<IStackStyles> = {
+      root: {
+        alignSelf: "flex-start",
+        border: "2px",
+        marginLeft: "3px",
+        marginRight: "3px",
+      },
+    }
+    const dateTimePickerStyles = mergeStyleSets({
+      iconContainer: {
+        marginLeft: 5,
+      },
+    })
+    const classNames = mergeStyleSets({
+      iconContainer: {
+        margin: "0px 0px",
+        height: 25,
+        width: "90px",
+      },
+    })
+  
+    const datePickerStyles: Partial<IDatePickerStyles> = { root: { maxWidth: 145, marginTop: 5, } };
+    const timePickerStyles: Partial<IDatePickerStyles> = { root: { maxWidth: 75, marginLeft: 5, marginRight: 5 } };
+    const onTimeChange = (event: React.FormEvent<IComboBox>, date: Date) => {        
+      var localDate = new Date()
+      localDate.setUTCFullYear(props.endDate.getFullYear(), props.endDate.getMonth(), props.endDate.getDate())
+      localDate.setUTCHours(date.getHours());
+      localDate.setUTCMinutes(date.getMinutes());
+      props.setEndDate(localDate)        
+     };
+    const onDateChange = (date: Date | null | undefined): void => { 
+      let localDate: Date = new Date();
+      localDate.setUTCFullYear(date!.getFullYear(), date!.getMonth(), date!.getDate())
+      localDate.setHours(props.endDate.getHours());
+      localDate.setMinutes(props.endDate.getMinutes());
+      props.setEndDate(localDate)  
+     };
```

Comment:
These two functions appear to be almost duplicates and could be combined.

Reply:
No actually there is a difference, here functions are given as props to different FluentUI component firstly and secondly, `onDateChange` only the date gets changed and on `onTimeChange` on the time gets changed the date being same so they are different.

Reply:
I don't think the change I  had in mind is worth pushing for here.